### ISSUE TITLE
Update a change in a related column

### ISSUE THREAD

**Issue by jurraburusvy** [Servoy/aggridcomponents#22]
Changes to foundset columns are also done to the db as usual, but there is no onDataChange event for related columns to attach a saveData. The only way I found is to add a button to saveData but this can cause multiple changes to be lost if the user does not click that button. I tried to run that method under onColumnDataChange event but it is not fired.

What is the best way to write related column changes into the db using Data Grid?

**Comment by seanthomasdevlin** [comment]
Hi Juan,

Can you clarify - you are editing a related data provider in the grid. And the data change event is not fired? Sounds like it could be a bug.

**Comment by jurraburusvy** [comment]
Hi Sean,
1) Working with Servoy 8.3.1 and NG-Grids, I place a Data Grid with a couple of columns from Orders and Company name from a relation from Orders to Customers.
I set as modifiable one of the columns of Orders and the Company Name from Customers.
I run and if I change the freight that is the modifiable column from the foundset of the form, the data is not changed into the db. I added a SaveData button but does not update the database. I also tried with the OnColumnDataChange event in the Grid but SaveData does not change the db.

2) Working with Servoy 8.3 and NG-Grids, when I modify a cell that belongs to the current foundset, the data is updated inmediately into the database. When I modify a cell from a related table, I need to have a SaveData Button, and then it saves it. The OnColumnDataChange event does not update the db.

In summary, it would be nice that the NG-Grids work like Table component where foundset data is updated when you finish or move to other record, and related data can be saved using SaveData (if you can update it the same as the foundset just moving to other record would be great). But in NG-Grids there is no OnDataChange event and the columndatachange does not work for this.

**Comment by jurraburusvy** [comment]
Hi Marius,

Please take a look at this short video showing the problem
https://youtu.be/ApmN0PENd2w 
AutoSave was not changed, so it is true

**Comment by MariusMunteanServoy** [comment]
Hi Juan,

You use servoy 8.3.0rc1(3201) and aggrid version 2.0.0rc which is supported only from 8.3.1 (build 3205). If you delete the zip and try to download this version of aggrid with WebPackageManager (right-click on the solution in solex) this will not appear in there. (1.0.2 will be latest).

**Comment by jurraburusvy** [comment]
Hi Marius,
The 8.3.1 rc1 we were using was downloaded (exe) from https://forum.servoy.com/viewtopic.php?f=1&t=22228 and it seems the installer in that link was not updated.
We just updated now to the released 8.3.1  and now works fine!

There´s just one important thing to improve. If you work with related columns based on a relation (orders_to_customers.company) and make that editable, that change is not saved to the database and there is not event that works for that. The only way is to add an external button or option to Save Data. Having an onDataChange for the column or the hole grid would be fine.

**Comment by jcompagner** [comment]
as far as i know that should be saved when you press somewhere on the screen
or change the selection on the main

it will not save right away when you just make changes and jump to the next colum, servoy never did change then

(and if you have autoSave == false then it will not save anyway in the above scenario's you need the save button

**Comment by jcompagner** [comment]
also grouping grid has an event for that that should be called:
https://github.com/Servoy/aggridcomponents/blob/master/aggrid/groupingtable/groupingtable.spec#L103

that oncolumndatachange should be called when you make a change (related or not)

**Comment by jurraburusvy** [comment]
That´s the case, I call SaveData at oncolumndatachange event but does not save it to the db. I need to click on the form, outside of the Grid to update. I think adding an event in the grid to do that would give more control to update the changes made by the user.

**Comment by gboros** [comment]
oncolumndatachange is called as soon as you go out of editing the cell (tab-out, click-out, enter), but
it is called before updating the data on the server, because it can return false, in which case it should not send anything; the callback do get the old/new values as params

**Comment by gboros** [comment]
ah, I see now, you want to call saveData in the callback? well yes, that is a bit too early to have the cell data already sent to the db ...

**Comment by jurraburusvy** [comment]
Hi Gabi,
The db is updated when you click outside the grid and also with Enter key (sometimes you need to click on another row, not just Enter). Tab inside the grid does not update. And using up and down keys would be nice to update too.
